src/java.base/share/classes/sun/security/ssl/SSLSocketImpl.java

```diff
@@ -1687,6 +1695,11 @@ private void handleException(Exception cause) throws IOException {
             }
         }
 
+        if (cause instanceof SocketException) {
+            conContext.teardownTransport(cause, alert, false);
+            throw (SocketException)cause;
+        }
+
         throw conContext.fatal(alert, cause);
```

Comment:
It might be not necessary to change the TransportContext by adding a new teardownTransport() method. It would be good to keep the fatal() behavior as if a fatal alter will be sent.  Maybe, the exception thrown by fatal() could be replaced with the socket exception, like:

```
if (cause instanceof SocketException) {
     try {
        conContext.fatal(alert, cause);
     } catch (Exception) {
        // Just delivering the fatal alert, re-throw the socket exception instead.
     } finally {
        throw (SocketException)cause;
     }
} else {
      throw conContext.fatal(alert, cause);
}
```

Reply:
Thank you for the feedback, I have updated as recommended. I could not add a throw in the finally block as this generates a warning. Instead i have done this

```
            try {
                conContext.fatal(alert, cause);
            } catch (Exception e) {
                // Just delivering the fatal alert, re-throw the socket exception instead.
            }

            throw (SocketException)cause;
```